lib/useKResponsiveWindow.js

```diff
@@ -0,0 +1,199 @@
+import { computed, onBeforeUnmount, onMounted, ref } from '@vue/composition-api';
+
+/**
+ * Returns window inner width and height
+ * @returns {Object} The window's inner width and height
+ */
+function windowMetrics() {
+  return {
+    width: window.innerWidth,
+    height: window.innerHeight,
+  };
+}
+
+/**
+ * Check if Nuxt is server side rendering
+ * @returns {Boolean}
+ */
+function isNuxtServerSideRendering() {
+  return process && process.server;
+}
+
+/**
+ * Add a media query
+ * @param {String} query
+ * @param {requestCallback} callback
+ * @returns {Object} The mediaQueryList, eventHandler, and stopListening
+ */
+function addMediaQuery(query, callback) {
+  if (isNuxtServerSideRendering()) {
+    return;
```

Comment:
Is it up to the consumer of this function to know when to handle an `undefined` return here, or should we be giving a consistent return type at least? If we return an empty object, the value can still be destructured, but will give undefined for each of the destructured values.

Reply:
I think the latter makes more sense

Reply:
Thinking this through again, the code was intended to be defence measure against execution of the function during ssr as suggested by @bjester